src/pages/AddMembersPage/NewMemberEntry.tsx

```diff
@@ -1,25 +1,99 @@
-import React from 'react';
+import assert from 'assert';
+import React, { useEffect, useState } from 'react';
 
-import { FieldValues, UseFormRegister } from 'react-hook-form';
+import { isAddress } from '@ethersproject/address';
+import { client } from 'lib/gql/client';
+import { zEthAddress, zUsername } from 'lib/zod/formHelpers';
+import {
+  Control,
+  FieldValues,
+  useController,
+  UseFormRegister,
+  UseFormSetValue,
+} from 'react-hook-form';
+import { z } from 'zod';
 
 import { Box, Flex, Text, TextField } from '../../ui';
 import { X } from 'icons/__generated';
 
 import NewMemberGridBox from './NewMemberGridBox';
 
+const FormValuesSchema = z.object({
+  newMembers: z.array(
+    z.object({
+      address: zEthAddress.or(z.literal('')),
+      name: zUsername.or(z.literal('')),
+      entrance: z.string(),
+    })
+  ),
+});
+type FormValues = z.infer<typeof FormValuesSchema>;
+
 const NewMemberEntry = ({
   onFocus,
   onRemove,
   register,
   index,
   error,
+  control,
+  setValue,
 }: {
   onFocus(): void;
   onRemove?(): void;
   register: UseFormRegister<FieldValues>;
   index: number;
   error?: { name?: string; address?: string };
+  control: Control<FormValues>;
+  setValue: UseFormSetValue<FormValues>;
 }) => {
+  const [isFetched, setIsFetched] = useState<boolean>(false);
+
+  const { field: addressField, fieldState: addressFieldState } = useController({
+    control,
+    name: `newMembers.${index}.address`,
+    defaultValue: '',
+  });
+
+  useEffect(() => {
+    if (!addressFieldState.error && isAddress(addressField.value)) {
+      const getName = async () => {
+        try {
+          const { profiles } = await client.query(
+            {
+              profiles: [
+                {
+                  where: { address: { _ilike: addressField.value } },
+                  limit: 1,
+                },
+                { name: true },
+              ],
+            },
+            { operationName: 'NewMemberEntry_getUserName' }
+          );
+          assert(profiles, 'failed to fetch user profiles');
+
+          const name = profiles[0]?.name ?? '';
+          if (name.length > 0) {
+            setValue(`newMembers.${index}.name`, name);
```

Comment:
@sshmm when we do `setValue` we also need to clear out any errors on this form field...

Reply:
done